crates/node/src/shell/queries.rs

```diff
@@ -15,19 +15,24 @@ where
     /// the default if `path` is not a supported string.
     /// INVARIANT: This method must be stateless.
     pub fn query(&self, query: request::Query) -> response::Query {
-        let ctx = RequestCtx {
-            state: self.state.read_only(),
-            event_log: self.event_log(),
-            vp_wasm_cache: self.vp_wasm_cache.read_only(),
-            tx_wasm_cache: self.tx_wasm_cache.read_only(),
-            storage_read_past_height_limit: self.storage_read_past_height_limit,
-        };
-
         // Invoke the root RPC handler - returns borsh-encoded data on success
         let result = if query.path == RPC.shell().dry_run_tx_path() {
-            dry_run_tx(ctx, &query)
+            dry_run_tx(
+                unsafe { self.state.read_only().with_static_temp_write_log() },
```

Comment:
```suggestion
                // SAFETY: blah blah
                unsafe { self.state.read_only().with_static_temp_write_log() },
```
consider adding a small explanation on why this is fine